### ISSUE TITLE
[C++][Parquet] Minor: Remove "Experimental" for parquet::RecordReader

### ISSUE THREAD

**Issue by mapleFU** [apache/arrow#40868]
### Describe the enhancement requested

```c++
/// \brief Stateful column reader that delimits semantic records for both flat
/// and nested columns
///
/// \note API EXPERIMENTAL
/// \since 1.3.0
class PARQUET_EXPORT RecordReader 
```

RecordReader is added in arrow 1.3.0 in 2019. Now it's widely used as de-facto reader of parquet records. And Golang/arrow-rs also uses RecordReader. Should we remove the experimental here?

### Component(s)

C++, Parquet

**Comment by mapleFU** [comment]
cc @wgtmac @pitrou @emkornfield

**Comment by pitrou** [comment]
I agree that it's not experimental anymore.

**Comment by emkornfield** [comment]
I agree, I think we should also move it out of internal namespace which I think it is still in?

**Comment by pitrou** [comment]
Yes, we should probably do that too!

**Comment by thisisnic** [comment]
This issue has been closed because there has been no activity for the 14 days since the stale warning. Feel free to reopen if this has been closed in error.
